### PR TITLE
Fix mem leaks

### PR DESCRIPTION
**Pull request by scoopr**
These commits fix the remaining leaks that I could identify

### CODE REVIEW

**Comment by scoopr** [comment]
I suppose the objc-foundation removal code is the most invasive, and thinking about it, I guess it could be sidestepped with fixing the various setLabel calls with `&*nslabel` for parameter, and adding a `msg_send![nsname, release]` to `get_function`

**Comment by kvark** [comment]
Could you elaborate on why those dependencies are removed?

File: src/argument.rs [hunk 1]
```diff
@@ -140,8 +139,8 @@ foreign_obj_type! {
 impl StructMemberRef {
     pub fn name(&self) -> &str {
         unsafe {
-            let name: &NSString = msg_send![self, name];
-            name.as_str()
+            let name = msg_send![self, name];
+            crate::nsstring_as_str(name)
```
Comment (grovesNL) [inline src/argument.rs]: Is it possible to fix these upstream in objc-foundation instead of removing it?

**Comment by kvark** [comment]
If it's only about `NSString`, it seems like removing 2 external dependencies is a win, even if/when they themselves get the appropriate fixing.
I'm open to bringing them back when times comes, if @grovesNL thinks they are going to be useful.
bors r+

**Comment by grovesNL** [comment]
@kvark that sounds fine to me. objc-foundation also has `NSArray` etc. which would be good to define in a single crate IMO

**Comment by scoopr** [comment]
Yeah, I had removed the objc-foundation wrappers locally before I found a way to workaround the issue, but then felt that in this case less dependencies could be desirable, if it isn't used as much, though I'm ambivalent about it.

**Pull request merged**